### DISCUSSION TITLE
IMEXAMINE Issue with Latest Development Snapshot

### DISCUSSION THREAD

**Discussion by Don-Carona** [iraf-community/iraf#305] [Q&A] [answered]
macOS 13.4 (Ventura) on Apple Silicon (arm64); Tested on M1, M2, M2 Max and M2 Pro. XQuartz 2.8.5.

Compiling and installing the latest development snapshot (2.17.?) went very well. As a typical test I use imexamine to ensure xgterm is working properly and displaying the plots as expected. Using SAOImage DS9 macOS arm64 versions (8.4 - 8.5b1), imexamine apparently does not report the location of the cursor properly, and therefore, does not display the plot i.e.(radial profile) for the star, over which, the cursor is placed. I found that DS9 v8.3 (x86) works perfectly running under Rosetta II on Apple silicon. However, imexamine does not work properly using any of the most recent x86 versions of DS9 on Apple silicon.

It should be noted that under the stable version (2.17), imexamine works as expected. Based on this, I surmise that DS9 is not likely the issue.

Thanks to all for the excellent effort keeping IRAF alive. I create and maintain a number of programs for my astronomy students that leverage both IRAF and DS9 heavily. I am very appreciative of the fact you keep IRAF going.

Best Regards,
Don

**Comment by Don-Carona** [comment]
Update...this appears to be an issue in DS9...possibly related to the frame buffer. It is being looked into at present. If the FITS image is opened directly from DS9, there is no problem using imexamine or other tasks from any version.

Regards,
Don

**Reply by olebole** [reply]
OK. I haven't hat the opportunity to look into it (don't own an ARM Mac).

**Reply by Don-Carona** [reply]
SAOImage DS9 v8.3 works perfectly with IRAF in every respect. It's the later versions that appear to be problematic. This occurs on both ARM64 and x86_64 computers running macOS. The good folks at CfA are looking into the problem.

**Reply by Don-Carona** [reply, marked as answer]
Update...The issues related to IRAF, and especially IMEXAMINE, was indeed an “out-of-bounds” issue with SAOImage DS9 related to versions >= v8.4. A new official release will correct this.
